app/modules/enduser_setup.c

```diff
@@ -490,8 +520,15 @@ static void enduser_setup_http_urldecode(char *dst, const char *src, int src_len
       }
       *dst++ = c;
     }
+    if ((dst - dst_start) >= dst_len - 1)
```

Comment:
My bad, you're right. I must've lost a brace in the github folding or something. Apologies for the noise!